### PR TITLE
Add ability to switch modes

### LINKED ISSUE #9: As a Product Manager who needs clear organisation, I can switch between different modes
so that I know whether I am managing deliverables, contacts or meetings.

### PR DESCRIPTION
**Pull request by chrystalquek**
Close #9

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/mode/ModeParser.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+import seedu.address.logic.commands.mode.Command;
+import seedu.address.logic.commands.mode.SwitchCommand;
+import seedu.address.logic.commands.person.AddCommand;
+import seedu.address.logic.commands.person.HelpCommand; // take this out of Person
```
Comment (claraadora) [inline src/main/java/seedu/address/logic/parser/mode/ModeParser.java]: Maybe we can add a TODO keyword in the comment so it's easier for us to locate in the future
i.e. `// TODO take this out of Person `

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 2]
```diff
@@ -51,21 +54,25 @@
...
      */
-    public MainWindow(Stage primaryStage, LogicPerson logic) {
+    public MainWindow(Stage primaryStage, LogicMode logicMode, LogicPerson logicPerson) {
         super(FXML, primaryStage);
 
         // Set dependencies
         this.primaryStage = primaryStage;
-        this.logic = logic;
+        this.logicMode = logicMode;
+        this.logicPerson = logicPerson;
```
Comment (claraadora) [inline src/main/java/seedu/address/ui/MainWindow.java:65]: As mentioned above, we can discuss this soon :D But, I think we don't need to make any changes for this iteration 👍  

For a reminder, maybe we can add a TODO comment too here?

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 3]
```diff
@@ -106,17 +113,22 @@ private void setAccelerator(MenuItem menuItem, KeyCombination keyCombination) {
...
 
+    public void switchMode(ModeEnum mode) {
+        assert mode == null : "Mode should not be null";
+        this.mode = mode;
+    }
+
     /**
      * Fills up all the placeholders of this window.
      */
-    void fillInnerParts() {
```
Comment (claraadora) [inline src/main/java/seedu/address/ui/MainWindow.java:112]: Subject to the discussion above :D

File: src/main/java/seedu/address/ui/ModeEnum.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package seedu.address.ui;
```
Comment (claraadora) [inline src/main/java/seedu/address/ui/ModeEnum.java:1]: I think we can move this to the common directory? What do u think? :D

File: src/main/java/seedu/address/ui/UiManager.java [hunk 5]
```diff
@@ -22,15 +23,17 @@
     private static final Logger logger = LogsCenter.getLogger(UiManager.class);
     private static final String ICON_APPLICATION = "/images/address_book_32.png";
 
-    private LogicPerson logic;
+    private LogicMode logicMode;
+    private LogicPerson logicPerson;
```
Comment (claraadora) [inline src/main/java/seedu/address/ui/UiManager.java:27]: As mentioned above, we should discuss this 👍

**Review by claraadora — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 3 above)
Comment (chrystalquek) [reply src/main/java/seedu/address/ui/MainWindow.java:112]: I think this is for the initial initialisation of the Ui (to contain the Contacts List as of now). Can be changed later on to show dashboard

**Review by claraadora — APPROVED** [review]
LGTM

**Pull request merged**